tests/causal/cate_learning/test_meta_learners.py

```diff
@@ -0,0 +1,271 @@
+    )
+    expected = ["treatment_A", "treatment_B", "treatment_C"]
+
+    assert sorted(filtered) == sorted(expected)
+
+
+def test__filter_by_treatment():
+    values = [
+        [1.0, "treatment_A", 1],
+        [4.0, "treatment_C", 1],
+        [1.0, "treatment_B", 0],
+        [5.0, "treatment_A", 0],
+        [3.0, "control", 1],
+    ]
+
+    df = pd.DataFrame(data=values, columns=["feat1", "treatment", "target"])
+
+    selected_treatment = "treatment_A"
+
+    expected_values = [
+        [1.0, "treatment_A", 1],
+        [5.0, "treatment_A", 0],
+        [3.0, "control", 1],
+    ]
+
+    expected: DataFrame = pd.DataFrame(
+        data=expected_values, columns=["feat1", "treatment", "target"]
+    )
+
+    results = _filter_by_treatment(
+        df,
+        treatment_col="treatment",
+        treatment_name=selected_treatment,
+        control_name="control",
+    )
+
+    assert_frame_equal(results, expected)
+
+
+def test__create_treatment_flag():
+    df = pd.DataFrame(
+        {
+            "feature": [1.3, 1.0, 1.8, -0.1],
+            "group": ["treatment", "treatment", "control", "control"],
+            "target": [1, 1, 1, 0],
+        }
+    )
+
+    expected = pd.DataFrame(
+        {
+            "feature": [1.3, 1.0, 1.8, -0.1],
+            "group": ["treatment", "treatment", "control", "control"],
+            "target": [1, 1, 1, 0],
+            TREATMENT_FEATURE: [1.0, 1.0, 0.0, 0.0],
+        }
+    )
+
+    results = _create_treatment_flag(
+        df, treatment_col="group", treatment_name="treatment"
+    )
+
+    assert_frame_equal(results == expected)
+
+
+def test__fit_by_treatment():
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1, 0.0, 1.0, 2.2, 0.4, -5.0],
+            "x2": [10, 4, 15, 6, 5, 12, 14, 5, 12],
+            "treatment": [
+                "A",
+                "B",
+                "A",
+                "A",
+                "B",
+                "control",
+                "control",
+                "B",
+                "control",
+            ],
+            "target": [1, 1, 1, 0, 0, 1, 0, 0, 1],
+        }
+    )
+
+    learner_binary = logistic_classification_learner(
+        features=["x1", "x2", TREATMENT_FEATURE],
+        target="target",
+        params={"max_iter": 10},
+    )
+
+    treatments = ["A", "B"]
+
+    learners, logs = _fit_by_treatment(
+        df,
+        learner=learner_binary,
+        treatment_col="treatment",
+        control_name="control",
+        treatments=treatments,
+    )
+
+    assert len(learners) == len(treatments)
+    assert len(logs) == len(treatments)
+
+
+def test__predict_by_treatment_flag_positive():
+    df = pd.DataFrame(
+        {
+            "x1": [1.3, 1.0, 1.8, -0.1],
+            "x2": [10, 4, 15, 6],
+            TREATMENT_FEATURE: [1.0, 1.0, 0.0, 0.0],
+            "target": [1, 1, 1, 0],
+        }
+    )
+
+    learner_binary = logistic_classification_learner(
```

Comment:
Should we add a random_state here, since we are comparing specific values?

Reply:
I think we should mock the learner, instead. What do you think?

Reply:
I removed the values and now we are testing the input with 1 and 0 values only, since we don't need to test the learner.